### ISSUE TITLE
Cancelling the creation of a new queue causes a form validation error

### ISSUE THREAD

**Issue by PradyumnaShome** [illinois/queue#287]
**Steps to reproduce**
Create queue -> Cancel queue

**Expected Behavior**
Modal disappears

**Current Behavior**
- Form validation errors - "you must select a course, and enter a name", and then the modal disappears.
- Address bar has remnants of create queue action - "https://queue.illinois.edu/q/?course=&name=&location="

**Comment by nwalters512** [comment]
Good catch! Probably due to `type="submit"` on the cancel button in `components/NewQueue`. That blame falls solely on me 😢
